Assets/Tests/Runtime/ClientServer/ServerObjectManagerTests.cs

```diff
@@ -201,8 +201,10 @@ public void AddPlayerForConnectionAssetIdTest()
 
             connectionToClient.Identity = null;
 
-            Assert.That(serverObjectManager.AddCharacter(connectionToClient, playerReplacement, replacementGuid), Is.True);
-        }
+            serverObjectManager.AddCharacter(connectionToClient, playerReplacement, replacementGuid);
+
+            await AsyncUtil.WaitUntilWithTimeout(() => replacementIdentity == connectionToClient.Identity);
```

Comment:
shouldn't character be set right away? can we just assert it is set instead of await?

Reply:
didnt think that would work. turns out it does :). fixed